modules/specialized-guides/pages/salt/salt-connectivity.adoc

```diff
@@ -0,0 +1,156 @@
+[[salt-connectivity]]
+= Salt Connectivity
+
+Depending on the size and spreadness of the environment where {productname} is used some connectivity issues are possible.
+There are no common recommendations for all the possible use cases as the environments could be very different expecially if public clouds instances are involved.
+
+
+
+[[minions-connectivity]]
+== Minions Connectivity
+
+In case if clients are losing the connection to the {productname} Server (or {productname} Proxy if involved),
+they are unreachable from the {productname} Server web UI or using the CLI.
+To ensure if it's the case, it's better to check if the client is not reachable from the {productname} Server with [command]``salt MINION_ID test.ping``,
+while [command]``venv-salt-call test.ping`` or ([command]``salt-call test.ping`` in case if non-bundle salt is used on the client)
+is working fine on the client side, it is recommended to set `tcp_keepalive` parameters.
```

Comment:
Sentence is too long.  I'm not sure if I got it right.  Maybe, we can try sth like this:

```suggestion
In such a case, first check the connection from the {productname} Server with [command]``salt MINION_ID test.ping``.
[command]``venv-salt-call test.ping`` or ([command]``salt-call test.ping`` in case non-bundle {salt} is used on the client) is working fine on the client side, especially if `tcp_keepalive` parameters are set.
```

Reply:
Looks like the meaning has changed here. The idea here that the minion service can lose the connection while didn't get any notification about it, so minion is still holds dead tcp socket which can't be used and it's not trying to reconnect. To ensure if it's the case one command from the server should fail while the commands `...salt-call` from the client is not failing, and in this case it makes sense to add `tcp_keepalive...` parameters to the minion config to avoid such cases.

Reply:
I think I understand now want it is about.  Maybe, I'll come up with a proposal later.

Reply:
Next try:
```suggestion
To debug such a situation, check whether both communication directions between server and client are working.
On the server, check with [command]``salt MINION_ID test.ping`` that the server can reach the client.
If it fails, check on the client with [command]``venv-salt-call test.ping`` whether the server is reachable from the client.
If the communication from the client is working fine, it is recommended to set `tcp_keepalive` parameters on the client.

[NOTE]
====
In case the salt bundle is not used on the client, run [command]``salt-call test.ping`` instead of [command]``venv-salt-call test.ping``.
====
```
Maybe, we should also add the technical details about the dead TCP socket?  Maybe, instead of "communication" better the simpler term "connection"?

Reply:
Yes, is the most important thing here.  With the following proposal I try to keep the test scenario, and mostly only change the intro phrase.  Then I propose the solution in a short final sentence.  We could add the background info you provide, starting from "2 parallel channels with the same direction open. And the idea of the check is to ... etc.".

Or we just merge it as-is now, and then ask Joseph for help.